### PR TITLE
CLOUDSTACK-10057: listNetworkOfferings now returns the correct number  of offerings.

### PR DESCRIPTION
**Pull request by sgoeminn**
https://issues.apache.org/jira/browse/CLOUDSTACK-10057

### CODE REVIEW

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 1]
```diff
@@ -4788,7 +4788,7 @@ protected void validateNtwkOffDetails(final Map<Detail, String> details, final M
             // Now apply pagination
             final List<? extends NetworkOffering> wPagination = StringUtils.applyPagination(supportedOfferings, cmd.getStartIndex(), cmd.getPageSizeVal());
             if (wPagination != null) {
-                final Pair<List<? extends NetworkOffering>, Integer> listWPagination = new Pair<List<? extends NetworkOffering>, Integer>(wPagination, offerings.size());
+                final Pair<List<? extends NetworkOffering>, Integer> listWPagination = new Pair<List<? extends NetworkOffering>, Integer>(wPagination, supportedOfferings.size());
```
Comment (rhtyd) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: Additional review and testing requested, if we need to change this line, then 4798 may have similar code as well.

Comment (sgoeminn) [reply server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: Line 4798 is correct. The supportedOfferings list is build within the ` if (parseOfferings)` code block. In the else case we want to return the offerings list and the size of the offerings list (supportedOfferings is also not defined in that case).

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by fmaximus** [comment]
I've added an extra unit test for the changed code, to keep the history intact in relation to the Trillian Tests, I have not squashed yet. I will squash once we have approval.

**Comment by rhtyd** [comment]
@blueorangutan package

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java (hunk 1 above)
Comment (krissterckx) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: LGTM

**Comment by fmaximus** [comment]
@rhtyd Do you agree to merge this? I have squashed both commits, but somehow Jenkins fails to report back its state, and the build is gone (hooray for only keeping 5 PR builds).

**Comment by krissterckx** [comment]
@rhtyd @fmaximus Can this be merged in ? thanks

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java (hunk 1 above)
Comment (rhtyd) [reply server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: @sgoeminn why not use `wPagination.size()` ?

Comment (sgoeminn) [reply server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: @rhtyd we always want to return the number of total found offerings and not only the number of offerings in the page that is returned. This is the same behavior as all the other list cmds.

Comment (rhtyd) [reply server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: @sgoeminn thanks.

**Comment by rhtyd** [comment]
LGTM, sure @fmaximus @krissterckx we can merge this.

**Pull request merged**